### ISSUE TITLE
dropout is 0.0

### ISSUE THREAD

**Issue by dipsivenkatesh** [karpathy/nanoGPT#455]
The dropout of the GPT model in GPTConfig class is set to 0.0, this means there won't be any dropout when training though, correct?

**Comment by amar-jay** [comment]
Yes, by deafult

**Comment by muerghq** [comment]
Many recent researches on LLMs proved that it's ok to not do dropout in pertaining. But you normally want dropout in fine tuning to avoid overfitting.

**Comment by siddharthji07** [comment]
Yes, Absolutely core because dropout of 0.0 during training shows that there is no dropout during training
